mlir-tensorrt/executor/lib/Conversion/MemRefToExecutor.cpp

```diff
@@ -548,6 +550,43 @@ void executor::populateMemRefToExecutorPatterns(
 }
 
 namespace {
+
+class RemoveNoOpClonePattern : public OpRewritePattern<bufferization::CloneOp> {
+public:
+  using OpRewritePattern<bufferization::CloneOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(bufferization::CloneOp op,
+                                PatternRewriter &rewriter) const override {
+    if (op.getInput().getType() == op.getOutput().getType()) {
+      rewriter.replaceOp(op, op.getInput());
+      return success();
+    }
+    return failure();
+  }
+};
+
+class RemoveRedundantCastsPattern
```

Comment:
This should also be removed. Requiring this means something is wrong before running this pass.

Reply:
Sure. This is not required.